File: client/iih_stub/iih.ci_stub/iih.ci.mr/i/ICiMrOutQryServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using xap.mw.coreitf.d;
using xap.mw.core.data;

namespace iih.ci.mr.i
{
    public interface ICiMrOutQryServices
    {
        /// <summary>
        /// 根据就诊号获取现病史，如果没有数据则返回""
        /// </summary>
        /// <param name="idEnt"></param>
        /// <returns></returns>
        String GetCiMrHistoryOfPresentIllness(String idEnt);

        /// <summary>
        /// 根据就诊号获取最后一次病程FMap，如果没有数据则返回new FMap()
        /// </summary>
        /// <param name="idEnt"></param>
        /// <param name="isFirst"></param>
        /// <returns></returns>
        FMap2 GetCiMrCourseOfLastDisease(String idEnt, FBoolean isFirst);

        /// <summary>
        /// 根据就诊号获取所有门诊病历
        /// </summary>
        /// <param name="strIdEnts"></param>
        /// <returns></returns>
        FMap2 GetOPCiMrByIdEnts(String[] strIdEnts);
    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/i/ICiMrOutQryServices.java
package iih.ci.mr.i;

import xap.mw.core.data.BizException;
import xap.mw.core.data.FMap2;
import xap.mw.coreitf.d.FBoolean;

public interface ICiMrOutQryServices
{
	
	/**
	 * 根据就诊号获取现病史，如果没有数据则返回""
	 * @param idEnt
	 * @return
	 */
	public abstract String getCiMrHistoryOfPresentIllness(String idEnt)  throws BizException;

	/**
	 * 根据就诊号获取最后一次病程FMap，如果没有数据则返回new FMap();
	 * @param idEnt
	 * @return
	 * @throws BizException
	 */
	public abstract FMap2 getCiMrCourseOfLastDisease(String idEnt,FBoolean isFirst) throws BizException; 
	
	/**
	 * 根据就诊号获取所有门诊病历
	 * @param strIdEnts
	 * @return
	 * @throws BizException
	 */
	public abstract FMap2 getOPCiMrByIdEnts(String[] strIdEnts) throws BizException;
}
